src/Shared/Helpers/OutputBuilder/OutputBuilder.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.CodeAnalysis.Sarif;
+using System;
+
+namespace Microsoft.CST.OpenSource.Shared
+{
+    /// <summary>
+    /// Builds the output text based on the format specified
+    /// </summary>
+    public class OutputBuilder : IOutputBuilder
+    {
+        public enum OutputFormat
+        {
+            sarifv1,
+            sarifv2,
+            text // no sarif, just text
+        };
+
+        public OutputFormat? currentOutputFormat { get; private set; } = OutputFormat.text; // default = text
+        IOutputBuilder? currentBuilder;
+
+        public OutputBuilder(string format)
```

Comment:
I don't understand this pattern.  Your constructor here constructs an internal IOutputBuilder held inside of an IOutputBuilder so this class just holds the real output builder obscuring it from the user.  

This should either be a factory function, or you should switch on format outside and have constructors for each of the derived outputbuilders with this one as an abstract class.

Reply:
makes sense to change it into a factory, I spent too much time on this trying to avoid switch cases everywhere else other than just a single class...and turned it into a frankenstein without achieving the objective. Reverting back to a factory